orca-applications/src/main/groovy/com/netflix/spinnaker/orca/applications/tasks/DeleteApplicationTask.groovy

```diff
@@ -42,15 +47,17 @@ class DeleteApplicationTask extends AbstractFront50Task {
           if (re.response?.status == 404) {
             return TaskResult.SUCCEEDED
           }
-          log.error("Could not create or update application permission", re)
+          log.error("Could not delete application permission", re)
           return TaskResult.builder(ExecutionStatus.TERMINAL).outputs(outputs).build()
         }
+        // delete Managed Delivery data
+        keelService.deleteDeliveryConfig(application.name)
```

Comment:
`keelService` may be null here.

Reply:
So can `front50Service` in the existing code... 🙂 

TBH, I never understood why we follow this pattern of marking all beans not-required and then assuming they're available, but I didn't want to stray from the pattern for fear of there being context I just don't have. Let me add a check here.

Reply:
Halyard, I believe would launch a minimal dependencies `orca`. I'm not a fan tho! (as halyard is decommissioned these patterns should go away).